src/map/scaleselector.js

```diff
@@ -252,6 +252,28 @@ class ScaleselectorController {
     }
   }
 
+  /**
+   * Calculate a scale from a number representing an index of scales or
+   * from a number between two scales indexes.
+   * @param {number} zoom An int or float value representing an index of
+   *     scales or a value between two scales indexes.
+   * @return {number} A calculated scales value or undefined.
+   */
+  getCalculateScale(zoom) {
+    // if zoom is an index of scales, return the matching scale.
+    let scale = this.scales[zoom];
+    if (scale !== undefined) {
+      return scale;
+    }
+
+    // If zoom is not an exact index of scales, try to determine the current scales from the zoom value.
+    const flooredZoom = Math.floor(zoom);
+    const lowerScale = this.scales[flooredZoom];
+    const upperScale = this.scales[flooredZoom + 1];
+    scale = lowerScale - (lowerScale - upperScale) * (zoom - flooredZoom);
+    return isNaN(scale) ? undefined : Math.round(scale);
```

Comment:
Why not
```suggestion
    const inchesPerMeter = 39.37;
    const dpi = 25.4 / 0.28;
    return olView.getResolution() * inchesPerMeter * dpi;
```
Then it will also work with an empty scales :-)
See also: https://github.com/openlayers/openlayers/blob/2dd9342a982e14c068cdd8c6492d2c6f79e8f81f/src/ol/control/ScaleLine.js#L463-L473

Reply:
I prefer my solution, I don't depend on hard-coded value (like dpi - even if these values should not changes) and it's easy to test.
But I can try to adapt the code. Do you want that I change ? I can't do it immediate (I've no time to do it right now).

Reply:
\+ it looks not working as expected, the scale value obtained is more than the double of the real scale. But I've not tried to know why.

Reply:
I just verify and It's the current values that are completely wrong... => new proposal